dbms/src/Storages/DeltaMerge/SSTFilesToBlockInputStream.cpp

```diff
@@ -64,20 +64,97 @@ SSTFilesToBlockInputStream::~SSTFilesToBlockInputStream() = default;
 
 void SSTFilesToBlockInputStream::readPrefix()
 {
-    for (UInt64 i = 0; i < snaps.len; ++i)
+    bool use_multi = false;
+    if (snaps.len > 3)
     {
-        const auto & snapshot = snaps.views[i];
-        switch (snapshot.type)
+        use_multi = true;
+    }
```

Comment:
If TiKV send 2 default cf file and 1 write cf file (can it?) and make this assumption break? Maybe always use `MultiSSTReader` for reading?

Reply:
This can actually happen, so we will check again if `snaps.len <= 3`.
MultiSSTReader can be always used though.

Reply:
If we always use MultiSSTReader, then I think
* We can eliminate some unnecessary branches, which make the code more clear and testable
* Make `SSTReader` a pure interface

Reply:
I think we'd better not move all logic of `SSTReader` into `MultiSSTReader` and make it pure virtual, since SSTReader reads an sst file, which is quite cohesive. Meanwhile, there are already some tests either based on or directly test `SSTReader`.

Reply:
I create a new `MonoSSTReader` for `SSTReader`, and then make `SSTReader` an abstract class, how about this? @JaySon-Huang